src/engine/QueryExecutionContext.h

```diff
@@ -112,16 +118,22 @@ class QueryExecutionContext {
     return _sortPerformanceEstimator;
   }
 
-  void readCostFactorsFromTSVFile(const string& fileName) {
-    _costFactors.readFromFile(fileName);
-  }
-
   [[nodiscard]] double getCostFactor(const string& key) const {
     return _costFactors.getCostFactor(key);
   };
 
   ad_utility::AllocatorWithLimit<Id> getAllocator() { return _allocator; }
 
+  void signalQueryUpdate(const RuntimeInformation& runtimeInformation) {
+    webSocketManager_.addQueryStatusUpdate(
+        owningQueryId_.toQueryId(),
+        nlohmann::ordered_json(runtimeInformation).dump());
+  }
+
+  ~QueryExecutionContext() {
+    webSocketManager_.releaseQuery(owningQueryId_.toQueryId());
```

Comment:
This is currently a little bit dangerous. If the webSocketManager has no io context the destructor throws